packages/ui/src/helper.ts

```diff
@@ -361,8 +360,10 @@ export const fmtTemplate = (template: Template, schemasList: SchemaForUI[][]): T
         delete cur.id;
         // @ts-ignore
         delete cur.key;
-        // @ts-ignore
-        delete cur.data;
+        if (!cur.readOnly) {
+          // @ts-ignore
+          delete cur.content;
+        }
```

Comment:
hmm so you can only use `content` if it's readOnly? 🤔 

I'm not sure how this will work for the use-case of content added by the editor and inputs, such as my case.

I was intending to use `data` for the JSON array of variables and `content` the editors text. 

I will think about it more tomorrow.

Reply:
Ah, indeed. We could consider removing `sampledata` and instead always use `content`. This way, `content` can be used whether it's `readOnly` or not. It's also feasible to set `propPanel`'s `defaultValue` in the schema to use `content`.

Of course, since `sampledata` will be eliminated, this would be a breaking change, but I've been thinking of eventually removing `sampledata` and `columns`.

Regardless of whether we actually implement it, there's potential for further simplification, so I'll try modifying it to use `content` instead of `sampledata`.

Reply:
OK. 

How would that work for my case where there's both `content` and `inputs` at the same time? 

![content-inputs-data](https://github.com/pdfme/pdfme/assets/7068515/aef4f744-a906-4e98-8136-df722dba6027)

Currently, `schema.data` models the inputs but you have consolidated these together above.

Reply:
In your case, input/data is actually a property of the schema. It cannot be substituted for the inputs passed to the generator.

In my understanding, placeholders are automatically replaced by the customer's data, but how exactly is the `Input/Data` in the `propPanel` configured?

Reply:
For the `propPanel` I was going to set the sample values added on `schema.data`. 
These would then be available as `sampledata` which could then  used for substitution on both the editor UI when not in editing mode, and when previewing the PDF.

I guess my issue is that `content` != `input`, i need both to be separate, and I think others writing plugins might need this in future. 

I think there are 3 cases:

* `content` == `input`  (pdfme original behaviour)
* `content` with no input (ready only version)
* `content` && `input` (a combination of both together)

I think any redesign should consider all of these cases

Reply:
Thank you for the clear explanation.

To keep it simple while addressing the three cases, I think a custom preprocessing of the generator function is necessary. https://github.com/pdfme/pdfme/pull/388#issuecomment-1868248936

Don't you think this approach is good?